en/components/calendar.md

```diff
@@ -4,40 +4,190 @@ _description: Create an intuitive calendar for applications to display date info
+
 ```html
-<igx-calendar [(ngModel)]="myDateValue"></igx-calendar>
+<!-- app.component.html -->
+<igx-calendar #calendar
+    ...
+    selection="range"
+    (onSelection)="verifyRange($event)">
+</igx-calendar>
 ```
-<div class="divider--half"></div>
+
+The value passed on in the **onSelection** event is the collection of dates selected, so we can read its length to base our logic upon it. If we alert the user for the invalid selection, we also reset the selection to contain only the first date from the range using the **selectDate** method:
+
+```typescript
+// app.component.ts
+...
+public verifyRange(dates: Date[]) {
+    if (dates.length > 5) {
+        this.calendar.selectDate(dates[0]);
+        this.dialog.open();
+    }
+}
+```
+
+Let's try this out by playing around with selecting ranges:
+<div class="sample-container" style="height: 460px">
+    <iframe id="calendar-sample-3-iframe" src='{environment:demosBaseUrl}/calendar-sample-3' width="100%" height="100%" seamless frameBorder="0" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<div>
+    <button data-localize="stackblitz" class="stackblitz-btn" data-iframe-id="calendar-sample-3-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+
+### Templating
+
+We have seen how to make use of the **IgxCalendarComponents** API (properties, events, methods) so that we configure the calendar per our requirements and interact with it programatically. Now we want to go further and customize its look, benefiting from the header and subheader templating capabilities.
+
+To do that we need to decorate a ng-template inside the calendar with **igxCalendarHeader** or **igxCalendarSubheader** directive and use the context returned to customize the way the date is displayed.
+The template decorated with the **igxCalendarHeader** directive is rendered only when the calendar selection is set to single. The **igxCalendarSubheader** is available in all selection modes.
+
+Our example slightly modify the default template and will make the header display the full date and modify the subheader to include the weekday:
+
+```html
+<!-- app.component.html-->
+<igx-calendar>
+    <!-- Modify the header to display the month (in titlecase), day and weekday -->
+    <ng-template igxCalendarHeader let-parts>
+        {{ parts.month.combined | titlecase }} {{parts.day.combined }} {{ parts.weekday.combined }}
+    </ng-template>
+    <ng-template igxCalendarSubheader let-parts>
+        <span class="date__el" (click)="parts.monthView()">{{ parts.month.combined }}</span>            
+        <span class="date__el" (click)="parts.yearView()">{{ parts.year.combined }}</span>
+    </ng-template>
+</igx-calendar>  
+```
+For more detailed explanation of what are **parts** in the template context, please refer to **TemplateContext** under the [API](http://localhost:3000/components/calendar.html#template-context) section. Let's see how the templated calendar looks like:
```

Comment:
"Let's see what the templated calendar looks like" or "Let's see how the templated calendar looks"